src/cirrus/lib2/eventdb.py

```diff
@@ -0,0 +1,144 @@
+import logging
+import os
+from datetime import datetime
+from enum import Enum, unique
+from typing import Any, Dict, Optional
+
+import boto3
+
+logger = logging.getLogger(__name__)
+
+
+@unique
+class StateEnum(Enum):
+    PROCESSING = "PROCESSING"
+    COMPLETED = "COMPLETED"
+    FAILED = "FAILED"
+    INVALID = "INVALID"
+    ABORTED = "ABORTED"
+
+
+class EventDB:
+    def __init__(self, event_db_and_table_names: Optional[str] = None):
+        self.tsw_client = boto3.client("timestream-write")
+        self.tsq_client = boto3.client("timestream-query")
+
+        if event_db_and_table_names is None:
+            event_db_and_table_names = os.getenv("CIRRUS_EVENT_DB_AND_TABLE")
+        if not event_db_and_table_names:
+            raise Exception("Event DB and table name not configured.")
+        db_and_table_names_array = event_db_and_table_names.split("|")
+        self.event_db_name = db_and_table_names_array[0]
+        self.event_table_name = db_and_table_names_array[1]
+
+    def write_timeseries_record(
+        self,
+        key: Dict[str, str],
+        state: StateEnum,
+        event_time: str,
+        execution_arn: Optional[str] = None,
+    ) -> Any:
+        parts = key.get("collections_workflow", "").rsplit("_", 1)
+        if not len(parts) == 2:
+            logger.error(
```

Comment:
Is there a reason to simply log an error vs raising an exception? It seems like it should be up to the caller to determine how to handle a failure case.

Same goes for other places with similar log and return behavior.

Reply:
I tend to prefer a style where Exceptions aren't thrown. In this case, if something is wrong with the parameters provided to the eventdb, I don't want it to possibly propagate an error out and cause the workflow to fail -- I'd rather miss the event and let the workflow succeed.

Reply:
Well, now instead of just logging the failures in the the statedb tests, they fail, so I'm going to have to figure that out out lol. (I probably just need to mock in the service with moto)

Reply:
ValueErrors added, tests fixed